### PR TITLE
Append data-param-* attributes to amp-iframe src as query params.

### LINKED ISSUE #1209: Consider sending data- attributes to amp-iframe
This would make certain use case more idiomatic, instead of using programmatic URL construction. Basically we'd just add all the data-attributes to the URL as query string arguments.

At this point I'm not 100% convinced it is a good idea and it is also less ideal for more advanced parameters. One use case is passing CSS into the iframe, which is still awkward via a URL argument.

### PR DESCRIPTION
**Pull request by mkhatib**
Closes #1209

### CODE REVIEW

File: extensions/amp-iframe/0.1/amp-iframe.js [hunk 1]
```diff
@@ -274,6 +276,26 @@ export class AmpIframe extends AMP.BaseElement {
...
+   */
+  getDataParams_() {
+    const attributes = this.element.attributes;
+    const params = {};
+    for (let i = 0; i < attributes.length; i++) {
+      const attr = attributes[i];
+      const matches = attr.nodeName.match(/^data-param-(.+)/);
+      if (matches) {
+        const param = matches[1];
+        params[param] = attr.nodeValue;
```
Comment (cramforce) [inline extensions/amp-iframe/0.1/amp-iframe.js]: In the ads case, where we have something similar we run `dashToCamelCase` over the param names. What do you think, does that make sense here?
https://github.com/ampproject/amphtml/blob/873385d7f745f0c119cda6c9b665f75b7e350873/src/string.js#L22

Both options suck: With it you cannot make dashes and without it, everything will be lower case.

File: extensions/amp-iframe/amp-iframe.md [hunk 2]
```diff
@@ -51,6 +51,10 @@ Note also, that the sandbox applies to all windows opened from a sandboxed ifram
 
 See the [the docs on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attr-sandbox) for further details on the sandbox attribute.
 
+##### data-param-*
+
+All `data-param-*` attributes will be added as query parameter to the iframe src. For example, you can add `data-param-autoplay=1` to a YouTube amp-iframe and it'll pass `?autoplay=1` as part of the iframe src.
```
Comment (cramforce) [inline extensions/amp-iframe/amp-iframe.md]: Please add a short example.

**Comment by cramforce** [comment]
Quick comment, otherwise looks great

File: extensions/amp-iframe/0.1/amp-iframe.js (hunk 1 above)
Comment (mkhatib) [inline extensions/amp-iframe/0.1/amp-iframe.js]: Makes sense. I think we should stay consistent of how we treat data attributes so calling `dashToCamelCase` on them sgtm.

File: extensions/amp-iframe/amp-iframe.md (hunk 2 above)
Comment (mkhatib) [inline extensions/amp-iframe/amp-iframe.md]: Done

**Comment by mkhatib** [comment]
This is all squashed and cleaned up. @cramforce let me know if you have other comments on this one.

**Comment by jridgewell** [comment]
Ping @cramforce. This LGTM.

File: extensions/amp-iframe/0.1/test/test-amp-iframe.js [hunk 3]
```diff
@@ -523,4 +537,5 @@ describe('amp-iframe intersection', () => {
       });
     });
   });
+
```
Comment (sriramkrish85) [inline extensions/amp-iframe/0.1/test/test-amp-iframe.js]: nit: remove line

**Comment by sriramkrish85** [comment]
LGTM -except for the nit

File: extensions/amp-iframe/0.1/test/test-amp-iframe.js (hunk 3 above)
Comment (mkhatib) [inline extensions/amp-iframe/0.1/test/test-amp-iframe.js]: Done

**Comment by rudygalfi** [comment]
Looks like this is ready to be merged?

File: examples/iframe.amp.html [hunk 4]
```diff
@@ -0,0 +1,64 @@
+<!doctype html>
+<html ⚡>
+<head>
+    <meta charset="utf-8">
+    <title>Facebook examples</title>
+    <link rel="canonical" href="amps.html" >
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+    <link href='https://fonts.googleapis.com/css?family=Questrial' rel='stylesheet' type='text/css'>
+    <!--<script async custom-element="amp-iframe" src="https://cdn.ampproject.org/v0/amp-iframe-0.1.js"></script>-->
```
Comment (cramforce) [inline examples/iframe.amp.html]: This is correct. Please change back.

`examples.build/iframe.amp.max.html` automatically generates the local version.

File: extensions/amp-iframe/0.1/amp-iframe.js [hunk 5]
```diff
@@ -275,6 +278,26 @@ export class AmpIframe extends AMP.BaseElement {
     }
     this.attemptChangeHeight(newHeight);
   }
+
+  /**
+   * Builds a key-value object representing data-params-* attributes on the
+   * element.
+   * @returns {Object}
```
Comment (erwinmombay) [inline extensions/amp-iframe/0.1/amp-iframe.js]: `@return {!Object}

File: examples/iframe.amp.html (hunk 4 above)
Comment (mkhatib) [inline examples/iframe.amp.html]: Done.

File: extensions/amp-iframe/0.1/amp-iframe.js (hunk 5 above)
Comment (mkhatib) [inline extensions/amp-iframe/0.1/amp-iframe.js]: Done

**Comment by mkhatib** [comment]
Done.

File: examples/iframe.amp.html [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            line-height: 2;
+            max-width: 400px;
+        }
+
+        /* For demo sake, push the iframes down to satisfy amp-iframe rules. */
+        .push-iframes-down {
+            min-height: 75vh;
+        }
+
+    </style><noscript><style>body {opacity: 1}</style></noscript>
```
Comment (erwinmombay) [inline examples/iframe.amp.html]: should we update this with the new boilerplate? /cc @sriramkrish85

**Pull request closed without merging**